File: src/Python/201.BitwiseANDOfNumbersRange.py

class Solution:
    def rangeBitwiseAnd(self, m, n):
        count = 0
        while m != n:
            m >>= 1
            n >>= 1
            count += 1
        return m << count


if __name__ == "__main__":
    solution = Solution()
    print(solution.rangeBitwiseAnd(5, 7))

File: src/CSharp/201.BitwiseANDOfNumbersRange.cs
namespace PeefyLeetCode.RangeBitwiseAnd
{
public class Solution {
    public int RangeBitwiseAnd(int m, int n) {
        int count = 0; 
        while (m != n){
            m >>= 1;
            n >>= 1;
            count++;
        }
        return n << count;
    }
}
}